docs/arcaflow/workflows/flow-control.md

```diff
@@ -107,3 +107,140 @@ outputs:
     my_plugin: !expr $.steps.my_plugin.outputs.success
     my_other_plugin: !expr $.steps.my_other_plugin.outputs.success
 ```
+
+### Reduce Repetition with `bindConstants()`
+
+The builtin function [bindConstants()](expressions.md#functions) allows you to 
+avoid repeating input variables for a foreach subworkflow. In the example 
+below, the input variable `name`'s value is repeated across each iteration in 
+this input. This requires a more repetitive input and schema definition. This 
+section will show you how to simplify it. 
+
+```yaml title="input-repeat-name.yaml"
+iterations:
+  - loop_id: 1
+    repeated_inputs:
+      name: mogo
+      ratio: 3.14
+  - loop_id: 2
+    repeated_inputs:
+      name: mogo
+      ratio: 3.14
+  - loop_id: 3
+    repeated_inputs:
+      name: mogo
+      ratio: 3.14
+  - loop_id: 4
+    repeated_inputs:
+      name: mogo
+      ratio: 3.14
+```
+
+```yaml title="workflow.yaml"
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties:
+        iterations:
+          type:
+            type_id: list
+            items:
+              id: SubRootObject
+              type_id: ref
+              namespace: $.steps.foreach_loop.execute.inputs.items
+            
+steps:
+  foreach_loop:
+    kind: foreach
+    items: !expr $.input.iterations
+    workflow: subworkflow.yaml
+    parallelism: 1
+
+outputs:
+  success:
+    fab_four: !expr $.steps.foreach_loop.outputs.success.data
+```
+
+```yaml title="subworkflow.yaml"
+version: v0.2.0
+input:
+  root: SubRootObject
+  objects:
+    SubRootObject:
+      id: SubRootObject
+      properties:
+        loop_id:
+          type:
+            type_id: integer
+        repeated_inputs:
+          type:
+            type_id: ref
+            id: RepeatedInputs
+    RepeatedInputs:
+      id: RepeatedInputs
+      properties:
+        name:
+          type:
+            type_id: string
+        ratio:
+          type:
+            type_id: float            
+
+steps:
+  example:
+    plugin:
+      deployment_type: image
+      src: quay.io/arcalot/arcaflow-plugin-template-python:0.4.0
+    input:
+      name: !expr $.input.repeated_inputs.name
+
+outputs:
+  success:
+    loop_id: !expr $.input.loop_id
+    ratio: !expr $.input.repeated_inputs.ratio
+    beatle: !expr $.steps.example.outputs.success
+```
```

Comment:
A nit-picks for this section. It may make sense to have a clear header below the paragraph about bindconstants that states that the workflow shown is the before workflow, (using ####). Otherwise, a reader may glance at it and not notice that it's an example of the problem.

Reply:
added a title